qdrant-landing/content/articles/sparse-vectors.md

```diff
@@ -0,0 +1,396 @@
+---
+title: "Sparse Vectors in Qdrant: Pure Vector-based Hybrid Search"
+short_description: "Sparse vectors are several times more efficient than dense vectors, making them a great choice for large-scale systems. They're also interpretable, which is a huge advantage over dense vectors."
+description: "Sparse vectors are a representation where each dimension corresponds to a word or subword, greatly aiding in interpreting document rankings. This clarity is why sparse vectors are essential in modern search and recommendation systems, offering an advantage over embedding or dense vectors."
+social_preview_image: /articles_data/sparse-vectors/social_preview.png
+small_preview_image: /articles_data/sparse-vectors/sparse-vectors-icon.svg
+preview_dir: /articles_data/sparse-vectors/preview
+weight: -40
+author: Nirant Kasliwal
+author_link: 
+date: 2023-12-03T13:00:00+03:00
+draft: false
+keywords:
+  - sparse vectors
+  - SPLADE
+  - hybrid search
+  - vector search
+---
+
+Think of a library with a vast index card system. Each index card only has a few keywords marked out (sparse vector) of a large possible set for each book (document). This is what sparse vectors enable for text. 
+
+## What is a Sparse Vector?
+
+Sparse vectors are like the Marie Kondo of data—keeping only what sparks joy (or relevance, in this case). 
+
+Consider a simplified example of 2 documents, each with 200 words. A dense vector would have several hundred non-zero values, whereas a sparse vector would have only 20 non-zero values.
+
+```python
+dense = [0.2, 0.3, 0.5, 0.7.....] # several hundred floats
+sparse = [{331: 0.5}, {14136: 0.7}] # 20 key value pairs
+```
+
+The numbers 331 and 14136 map to specific tokens in the vocabulary e.g. `['chocolate', 'icecream']`. The rest of the values are zero. This is why it's called a sparse vector.
+
+The tokens aren't always words though, sometimes they can be sub-words: `['ch', 'ocolate]` too.
+
+They're pivotal in information retrieval, especially in ranking and search systems. BM25, a standard ranking function used by search engines like [Elasticsearch](https://www.elastic.co/blog/practical-bm25-part-2-the-bm25-algorithm-and-its-variables?utm_source=qdrant&utm_medium=website&utm_campaign=sparse-vectors&utm_content=article&utm_term=sparse-vectors), exemplifies this. BM25 calculates the relevance of documents to a given search query. 
+
+BM25's capabilities are well proven, but it has its limitations. For example, it often needs building complex dictionary of synonyms, that is where sparse vectors come in. Sparse vectors excel in handling large text data, making them crucial in modern data processing and an improvement over more well-known methods like BM25.
```

Comment:
As far as I know, BM25 is just one of the sparse vector methods. This paragraph sounds as if it was something different, and sparse vectors were the alternative to it. Sparse vectors, in general, will struggle with synonyms. It might be different for SPLADE since a language model is used, but I'm unfamiliar with the concept.